src/device-feature.js

```diff
@@ -0,0 +1,16 @@
+export default class {
+
+    constructor(params) {
+        this.df_name = params['df_name'];
```

Comment:
這裡似乎不會測試 空值？

這裡丟 error 好了，自己定義一個 `ArguementError` 然後 `throw`

```suggestion
        this.df_name = params['df_name'];
        if (!this.df_name) {
            throw ArgumentError();
        }
```

Reply:
Python 是用 function 參數，所以沒給值會 error，但 JS 看起來沒檢查